### PR TITLE
guestfs: Add scripts for building qcow2 base image based on guestfs

### PR DESCRIPTION
**Pull request by vadorovsky**
Add base VM image provisioning which uses virt-customize from guestfs tools.
This is an attempt to try to replace Vagrant with terraform(-provider-libvirt),
since Packer came out to be not the best fit for modifying qcow2 images or
libvirt boxes.

Ref: #23
Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

### CODE REVIEW

File: contrib/guestfs/provision.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+tar zxvf crictl-$CRI_TOOLS_VERSION-linux-amd64.tar.gz -C /usr/local/bin
+rm -f crictl-$CRI_TOOLS_VERSION-linux-amd64.tar.gz
+
+# Use vanilla kubeadm instead of Kubic kubeadm.
+RELEASE="$(curl -sSL https://dl.k8s.io/release/stable.txt)"
+cd $DOWNLOAD_DIR
+curl -L --remote-name-all https://storage.googleapis.com/kubernetes-release/release/${RELEASE}/bin/linux/amd64/{kubeadm,kubelet,kubectl}
+chmod +x {kubeadm,kubelet,kubectl}
+
+RELEASE_VERSION="v0.9.0"
```
Comment (briandowns) [inline contrib/guestfs/provision.sh:67]: Would this ever need to be overridden at runtime? If so, maybe use an env var here?

**Comment by briandowns** [comment]
Nothing blocking for me other than thinking about some of the versioning in the provision script. Would there be any benefit to pass in a version at runtime and check if none present, set a default? Otherwise, LGTM.

File: contrib/guestfs/provision.sh (hunk 1 above)
Comment (nickgerace) [reply contrib/guestfs/provision.sh:67]: +1: even though this script has some self-declared hacks in it, I'm curious if we could put all the versions and parameters at the top as env vars

Comment (vadorovsky) [reply contrib/guestfs/provision.sh:67]: Or another idea - what about getting always the newest tag from https://github.com/kubernetes/release/?

**Comment by vadorovsky** [comment]
@briandowns @nickgerace Let me know if you are fine with the solution I've just pushed.

**Review by nickgerace — APPROVED** [review]

**Pull request merged**